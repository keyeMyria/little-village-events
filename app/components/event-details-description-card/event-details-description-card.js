import parseDescription from "./parse-description.js";

import DetailsCard from "../event-details-card";
import ExpandableView from "../expandable-view";

import HTMLView from "react-native-htmlview";

import { Text, StyleSheet } from "react-native";
import React from "react";


const styles = StyleSheet.create( {
    root: {
        flexDirection: "column",
        padding: 16,
        paddingBottom: 0
    },
    tab: {
        fontSize: 12,
        color: "#4a4a4a",
        paddingBottom: 16
    }
} );


const renderNode = node => {
    switch ( node.name ) {
        case "iframe": return null;
    }
};


const htmlStyles = StyleSheet.create( {
    body: {
        fontSize: 16,
    },
    strong: {
        fontWeight: "600"
    }
} );


export const DescriptionCard = ( { summary, description, details } ) =>
    <DetailsCard style={ styles.root }>
        <Text style={ styles.tab }>DESCRIPTION</Text>
        <ExpandableView initialMaxHeight={350} paddingBottom={18} >
            <HTMLView
                value={`<body>${summary ? `<p><strong>${summary}</strong></p>` : ""}${description}</body>`}
                stylesheet={ htmlStyles }
                renderNode={ renderNode }
            />
        </ExpandableView>
    </DetailsCard>
;




export default ( { event: { details } } ) => {
    const { summary, description } = details;
    if ( !summary && !description )
        return null;

    return <DescriptionCard summary={summary} {...parseDescription( description )} />;
}
