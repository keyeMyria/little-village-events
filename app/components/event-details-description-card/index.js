export { default } from "./event-details-description-card";
