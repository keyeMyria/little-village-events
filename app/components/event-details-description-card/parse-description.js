

export default desc => {
    const detailsRe = /(<p>(\w+(?:\s+\w+)*):(\s*\w+(?:\s+\w+)*)<\/p>)/g;

    const details = [];
    const description = desc
        .replace( detailsRe, ( match, p1, p2, p3 ) => {
            details.push( [ p2, p3 ] );
            return "";
        } )
        .replace( /(<br>)/g, "" )
        .replace( /(<p><\/p>)/g, "" )
        ;

    return { description, details };
};
