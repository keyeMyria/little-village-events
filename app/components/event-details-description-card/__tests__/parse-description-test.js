
import { parseDescription } from "../event-details-description-card";

test( "event-details-description-card/parseDescription", () => {
    expect( parseDescription( "plain text" ) ).toMatchSnapshot();
    expect( parseDescription( "<p>plain paragraph</p>" ) ).toMatchSnapshot();
    expect( parseDescription( "<p></p><p>ignore empty paragraphs</p><p><br></p>" ) ).toMatchSnapshot();

    expect( parseDescription( "<p>Chef Baroncini always rolls out terrific dough and amazing sauces. </p><br><p>More details to come!</p><br><p><br></p>" ) ).toMatchSnapshot();

    expect( parseDescription( "<p>Series:Coming Soon</p><p>Rating:R</p><p>Runtime:96 minutes</p><p>Director:Jason Reitman</p><p>Year Released:2018</p><p>Production Country:USA</p><p>Language:English</p><p>Showtimes announced Tuesday before opening date.</p><br><p>“Charlize Theron gives a fearless performance.\" - Owen Gleiberman, Variety</p><br><p>“Jason Reitman is back at his best again.\" - Alex Billington, First Showing</p><br><p>“A bitingly funny, deeply empathetic ode to motherhood.\"-\
        Tomris Laffly, Film Journal International</p><br><p>A new comedy from Academy Award nominated director Jason Reitman (Up in the Air) and Academy Award winning screenwriter and University of Iowa graduate Diablo Cody (Juno), Marlo (Charlize Theron), a mother of three including a newborn, is gifted a nanny by her brother (Mark Duplass). Hesitant to the extravagance at first, Marlo comes to form a unique bond with the thoughtful, surprising and sometimes challenging young nanny named\
        Tully.</p>" ) ).toMatchSnapshot();
} );
