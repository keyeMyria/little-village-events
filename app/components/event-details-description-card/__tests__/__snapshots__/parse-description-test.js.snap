// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`event-details-description-card/parseDescription extracts details 1`] = `
Object {
  "description": "<p>Showtimes announced Tuesday before opening date.</p><p>“Charlize Theron gives a fearless performance.\\" - Owen Gleiberman, Variety</p><p>“Jason Reitman is back at his best again.\\" - Alex Billington, First Showing</p><p>“A bitingly funny, deeply empathetic ode to motherhood.\\"-            Tomris Laffly, Film Journal International</p><p>A new comedy from Academy Award nominated director Jason Reitman (Up in the Air) and Academy Award winning screenwriter and University of Iowa graduate Diablo Cody (Juno), Marlo (Charlize Theron), a mother of three including a newborn, is gifted a nanny by her brother (Mark Duplass). Hesitant to the extravagance at first, Marlo comes to form a unique bond with the thoughtful, surprising and sometimes challenging young nanny named            Tully.</p>",
  "details": Array [
    Array [
      "Series",
      "Coming Soon",
    ],
    Array [
      "Rating",
      "R",
    ],
    Array [
      "Runtime",
      "96 minutes",
    ],
    Array [
      "Director",
      "Jason Reitman",
    ],
    Array [
      "Year Released",
      "2018",
    ],
    Array [
      "Production Country",
      "USA",
    ],
    Array [
      "Language",
      "English",
    ],
  ],
}
`;

exports[`event-details-description-card/parseDescription ignores empty paragraphs 1`] = `
Object {
  "description": "<p>ignore empty paragraphs</p>",
  "details": Array [],
}
`;

exports[`event-details-description-card/parseDescription ignores empty paragraphs 2`] = `
Object {
  "description": "<p>Chef Baroncini always rolls out terrific dough and amazing sauces. </p><p>More details to come!</p>",
  "details": Array [],
}
`;

exports[`event-details-description-card/parseDescription parses plain description 1`] = `
Object {
  "description": "plain text",
  "details": Array [],
}
`;

exports[`event-details-description-card/parseDescription parses plain description 2`] = `
Object {
  "description": "<p>plain paragraph</p>",
  "details": Array [],
}
`;
